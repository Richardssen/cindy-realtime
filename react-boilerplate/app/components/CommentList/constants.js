import { graphql } from 'react-apollo';
import gql from 'graphql-tag';
import UserLabel from 'graphql/UserLabel';

export const withCommentList = graphql(
  gql`
    query CommentListQuery(
      $orderBy: [String]
      $limit: Int
      $offset: Int
      $puzzle_Status_Gt: Float
      $puzzle_User: ID
      $user: ID
    ) {
      allCommentsLo(
        orderBy: $orderBy
        limit: $limit
        offset: $offset
        puzzle_Status_Gt: $puzzle_Status_Gt
        puzzle_User: $puzzle_User
        user: $user
      ) {
        edges {
          node {
            id
            user {
              ...UserLabel_user
            }
            puzzle {
              id
              title
              user {
                ...UserLabel_user
              }
            }
            content
            spoiler
          }
        }
        totalCount
      }
    }
    ${UserLabel}
  `,
  {
    options: ({ variables, fetchPolicy, page, itemsPerPage }) => ({
      variables: {
        limit: itemsPerPage,
        offset: itemsPerPage * (page - 1),
        ...variables,
      },
      fetchPolicy,
    }),
    props({ data }) {
      const { loading, allCommentsLo } = data;
      return {
        loading,
        allComments: allCommentsLo,
      };
    },
  },
);
