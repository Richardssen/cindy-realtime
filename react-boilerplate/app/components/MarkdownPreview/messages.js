/*
 * MarkdownPreview Messages
 *
 * This contains all the text for the MarkdownPreview component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  show: {
    id: 'app.components.MarkdownPreview.showPreview',
    defaultMessage: 'Show Preview',
  },
  hide: {
    id: 'app.components.MarkdownPreview.hidePreview',
    defaultMessage: 'Hide Preview',
  },
});
