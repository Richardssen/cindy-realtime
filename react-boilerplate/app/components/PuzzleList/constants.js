/*
 *
 * PuzzleList constants
 *
 */

export const DEFAULT_ACTION = 'app/PuzzleList/DEFAULT_ACTION';
