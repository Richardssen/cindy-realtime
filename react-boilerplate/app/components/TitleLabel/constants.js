/**
 *
 * TitleLabel constants
 *
 */

import classicJp from 'images/classicJp.svg';
import twentyQuestionsJp from 'images/twentyQuestionsJp.svg';
import littleAlbatJp from 'images/littleAlbatJp.svg';
import othersJp from 'images/othersJp.svg';
import yamiJp from 'images/yamiJp.svg';
import longtermYamiJp from 'images/longtermYamiJp.svg';

export const SHOW_PUZZLE = 'app/componnets/TitleLabel/SHOW_PUZZLE';

export const genreInfo = [
  {
    name: 'classic',
    pictureJp: classicJp,
  },
  {
    name: 'twentyQuestions',
    pictureJp: twentyQuestionsJp,
  },
  {
    name: 'littleAlbat',
    pictureJp: littleAlbatJp,
  },
  {
    name: 'others',
    pictureJp: othersJp,
  },
];

export const yamiInfo = [
  null,
  {
    name: 'yami',
    pictureJp: yamiJp,
  },
  {
    name: 'longtermYami',
    pictureJp: longtermYamiJp,
  },
];
