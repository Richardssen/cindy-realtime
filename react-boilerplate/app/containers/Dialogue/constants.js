/*
 *
 * Dialogue constants
 *
 */

export const DEFAULT_ACTION = 'app/Dialogue/DEFAULT_ACTION';
