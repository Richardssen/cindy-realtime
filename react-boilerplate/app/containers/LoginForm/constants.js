/*
 *
 * LoginForm constants
 *
 */

export const SHOW_MODAL = 'app/LoginForm/SHOW_MODAL';
