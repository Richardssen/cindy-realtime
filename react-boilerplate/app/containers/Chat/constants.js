/*
 *
 * Chat constants
 *
 */

import { componentsUserFragment } from 'containers/PuzzleActiveList/constants';

export const TOGGLE_MINICHAT = 'app/Chat/TOGGLE_MINICHAT';
export const TOGGLE_MEMO = 'app/Chat/TOGGLE_MEMO';
export const CHANGE_CHANNEL = 'app/Chat/CHANGE_CHANNEL';

export const CHANGE_TAB = 'app/Chat/CHANGE_TAB';
export const TABS = {
  TAB_CHAT: 'TAB_CHAT',
  TAB_CHANNEL: 'TAB_CHANNEL',
  TAB_DIRECTCHAT: 'TAB_DIRECTCHAT',
};

export const OPEN_MEMO = 'app/Chat/OPEN_MEMO';
export const CLOSE_MEMO = 'app/Chat/CLOSE_MEMO';

export const OPEN_MINICHAT = 'app/Chat/OPEN_MINICHAT';
export const CLOSE_MINICHAT = 'app/Chat/CLOSE_MINICHAT';

export const MINICHAT_CONNECT = 'ws/MINICHAT_CONNECT';
export const MINICHAT_DISCONNECT = 'ws/MINICHAT_DISCONNECT';

export const MINICHAT_ADDED = 'ws/MINICHAT_ADDED';
export const ADD_MINICHAT = 'app/Chat/ADD_MINICHAT';

export const MINICHAT_MORE = 'app/Chat/MINICHAT_MORE';
export const MORE_MINICHAT = 'app/Chat/MORE_MINICHAT';

export const INIT_MINICHAT = 'app/Chat/INIT_MINICHAT';

export const CHANGE_DIRECTCHAT = 'app/Chat/CHANGE_DIRECTCHAT';
export const SEND_DIRECTCHAT = 'ws/SEND_DIRECTCHAT';
export const DIRECTCHAT_RECEIVED = 'ws/DIRECTCHAT_RECEIVED';
export const ADD_DIRECTCHAT_MESSAGE = 'app/Chat/ADD_DIRECTCHAT_MESSAGE';

export const OPEN_CHAT = 'app/Chat/OPEN_CHAT';
export const OPEN_DIRECTCHAT = 'app/Chat/OPEN_DIRECTCHAT';

export const PublicChannels = ['lobby', '5-7-5'];

// {{{ const minichatFragment
const minichatFragment = `
  fragment components_minichat on MinichatNode {
    id
    content
    user {
      ...components_user
    }
  }
  ${componentsUserFragment}
`;
// }}}

// {{{ const minichatQuery
export const minichatQuery = `
  query($channel: String!) {
    allMinichats(channel: $channel, last: 10, orderBy: "id") {
      pageInfo {
        startCursor
        hasPreviousPage
      }
      edges {
        node {
          ...components_minichat
        }
      }
    }
  }
  ${minichatFragment}
`;
// }}}

// {{{ const minichatMoreQuery
export const minichatMoreQuery = `
  query($channel: String!, $before: String!) {
    allMinichats(channel: $channel, last: 10, before: $before, orderBy: "id") {
      pageInfo {
        startCursor
        hasPreviousPage
      }
      edges {
        node {
          ...components_minichat
        }
      }
    }
  }
  ${minichatFragment}
`;
// }}}

// {{{ const minichatUpdateQuery
export const minichatUpdateQuery = `
  query($id: ID!) {
    minichat(id: $id) {
      ...components_minichat
    }
  }
  ${minichatFragment}
`;
// }}}
