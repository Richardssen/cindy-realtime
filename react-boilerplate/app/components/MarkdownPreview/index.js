/**
 *
 * MarkdownPreview
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Panel, Button } from 'react-bootstrap';
import { text2md } from 'common';
// import styled from 'styled-components';

import { FormattedMessage } from 'react-intl';
import messages from './messages';

class MarkdownPreview extends React.Component {
  // eslint-disable-line react/prefer-stateless-function
  constructor(props) {
    super(props);
    this.state = {
      open: true,
    };

    this.togglePreview = this.togglePreview.bind(this);
  }

  togglePreview() {
    this.setState((prevState) => ({
      open: !prevState.open,
    }));
  }

  render() {
    return (
      <div>
        <Button onClick={this.togglePreview} block>
          {this.state.open ? (
            <FormattedMessage {...messages.hide} />
          ) : (
            <FormattedMessage {...messages.show} />
          )}
        </Button>
        <Panel collapsible expanded={this.state.open}>
          <div
            dangerouslySetInnerHTML={{
              __html: text2md(this.props.content, this.props.safe),
            }}
            style={{ overflow: 'auto' }}
          />
        </Panel>
      </div>
    );
  }
}

MarkdownPreview.propTypes = {
  content: PropTypes.string,
  safe: PropTypes.bool,
};

MarkdownPreview.defaultProps = {
  safe: false,
};

export default MarkdownPreview;
