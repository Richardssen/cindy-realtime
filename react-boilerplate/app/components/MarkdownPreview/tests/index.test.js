import React from 'react';
import { shallowWithIntl } from 'test/helpers';
import { TestMdText } from 'test/constants';

import { ButtonOutline as Button } from 'style-store';
import MarkdownPreview from '../index';

describe('<MarkdownPreview />', () => {
  const rendered = shallowWithIntl(<MarkdownPreview content={MDTEXT} />);

  it('Expect to render test markdown text properly', () => {
    expect(rendered.find('div')).toMatchSnapshot();
  });

  it('Expect to safely render test markdown text properly', () => {
    rendered.setProps({ safe: true });
    expect(rendered.find('div')).toMatchSnapshot();
  });

  it('Expect to collapse panel on button click', () => {
    expect(rendered.find('[pose="display"]')).toExist();
    rendered.find('[onClick]').simulate('click');
    expect(rendered.find('[pose="collapsed"]')).toExist();
  });
});
