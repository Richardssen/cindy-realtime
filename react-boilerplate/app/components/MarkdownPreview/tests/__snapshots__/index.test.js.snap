// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MarkdownPreview /> Expect to render test markdown text properly 1`] = `
Array [
  <div>
    <style-store__ButtonOutline
      onClick={[Function]}
      py={1}
      style={
        Object {
          "border": "1px solid #2075c7",
          "borderRadius": "0",
          "boxShadow": "inset 0 0 0 0",
        }
      }
      tabIndex="-1"
      w={1}
    >
      <FormattedMessage
        defaultMessage="Hide Preview"
        id="app.components.MarkdownPreview.hidePreview"
        values={Object {}}
      />
    </style-store__ButtonOutline>
    <MarkdownPreview__Panel
      pose="display"
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "
<h1>H1</h1>
<p>text <strong>bold</strong> <em>italic</em></p>
<blockquote>
<p>quote</p>
</blockquote>
<hr />
<ol>
<li>list
<ul>
<li>nested</li>
</ul>
</li>
</ol>
<p><font color=\\"red\\">red</font></p>
<div>right</div>
",
          }
        }
        style={
          Object {
            "overflow": "auto",
          }
        }
      />
    </MarkdownPreview__Panel>
  </div>,
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "
<h1>H1</h1>
<p>text <strong>bold</strong> <em>italic</em></p>
<blockquote>
<p>quote</p>
</blockquote>
<hr />
<ol>
<li>list
<ul>
<li>nested</li>
</ul>
</li>
</ol>
<p><font color=\\"red\\">red</font></p>
<div>right</div>
",
      }
    }
    style={
      Object {
        "overflow": "auto",
      }
    }
  />,
]
`;

exports[`<MarkdownPreview /> Expect to safely render test markdown text properly 1`] = `
Array [
  <div>
    <style-store__ButtonOutline
      onClick={[Function]}
      py={1}
      style={
        Object {
          "border": "1px solid #2075c7",
          "borderRadius": "0",
          "boxShadow": "inset 0 0 0 0",
        }
      }
      tabIndex="-1"
      w={1}
    >
      <FormattedMessage
        defaultMessage="Hide Preview"
        id="app.components.MarkdownPreview.hidePreview"
        values={Object {}}
      />
    </style-store__ButtonOutline>
    <MarkdownPreview__Panel
      pose="display"
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<style type=\\"text/css\\">
#test { flow: right; }
</style>
<h1>H1</h1>
<p>text <strong>bold</strong> <em>italic</em></p>
<blockquote>
<p>quote</p>
</blockquote>
<hr />
<ol>
<li>list
<ul>
<li>nested</li>
</ul>
</li>
</ol>
<p><font color=\\"red\\">red</font></p>
<div id=\\"test\\">right</div>
",
          }
        }
        style={
          Object {
            "overflow": "auto",
          }
        }
      />
    </MarkdownPreview__Panel>
  </div>,
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<style type=\\"text/css\\">
#test { flow: right; }
</style>
<h1>H1</h1>
<p>text <strong>bold</strong> <em>italic</em></p>
<blockquote>
<p>quote</p>
</blockquote>
<hr />
<ol>
<li>list
<ul>
<li>nested</li>
</ul>
</li>
</ol>
<p><font color=\\"red\\">red</font></p>
<div id=\\"test\\">right</div>
",
      }
    }
    style={
      Object {
        "overflow": "auto",
      }
    }
  />,
]
`;
